winch/codegen/src/isa/aarch64/masm.rs

```diff
@@ -316,7 +316,9 @@ impl Masm for MacroAssembler {
     }
 
     fn load_addr(&mut self, src: Self::Address, dst: WritableReg, size: OperandSize) -> Result<()> {
-        self.asm.uload(src, dst, size, TRUSTED_FLAGS);
+        let (base, offset) = src.unwrap_offset();
+        self.asm
+            .add_ir(u64::try_from(offset).unwrap(), base, dst, size);
```

Comment:
Perhaps we could call this masm method `compute_addr` to make it clearer that it's returning the address itself, not loading anything? (i.e., avoid the ambiguity where "load" could mean "load from memory" or "load a value into a register")

Reply:
Good point. Renamed, thanks!